### PR TITLE
Updated TreatmentSummary component by integrating it with PrintPreview Component #8430

### LINKED ISSUE #8430: Header and footer spacing in summaries
**Is your feature request related to a problem? Please describe.**
The format of all summaries in CARE needs to be improved. This includes
- discharge summaries
- treatment summaries
- Investigation summaries

The pages have no spacing at the header and footer.
The page breaks between sentences

**Describe the solution you'd like**

- [ ] Add spacing at the head and the foot of each page
- [ ] Do not allow page to break sentences. 

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Zeshanxviii**
Fixes [#8430 ]

## Proposed Changes

- Fixes #8430 
- Change 1
  Integrate Treatment summary component with printPreview component
- Change 2
  added CSS to global style  "CSS block ensures that when the PrintPreview component is printed, it has a clean, well-organized layout with proper spacing and separation of content, making it more suitable for a printed format."

@ohcnetwork/code-reviewers

printed treatmentSummary sample 
[NewtreatmentSUmmary.pdf](https://github.com/user-attachments/files/16824132/NewtreatmentSUmmary.pdf)

## Merge Checklist

- [x] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [x] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

File: src/style/index.css [hunk 1]
```diff
@@ -1306,4 +1306,39 @@ input[type="number"] {
...
\ No newline at end of file
+}
+
+
+
+@media print {
+    body {
+      padding-top: 40px;
+      padding-bottom: 40px;
+    }
```
Comment (rithviknishad) [inline src/style/index.css]: This padding is already applied by the `PrintPreview` component isn't it?

File: src/style/index.css [hunk 2]
```diff
@@ -1306,4 +1306,39 @@ input[type="number"] {
     font-weight: 700;
     color: #2856ff;
     font-size: 1rem;
-}
\ No newline at end of file
+}
+
+
+
+@media print {
```
Comment (rithviknishad) [inline src/style/index.css]: Would it be possible to use tailwind's `print:*` modifier to apply these styles coming under `@media print` without using CSS?

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/style/index.css (hunk 2 above)
Comment (Zeshanxviii) [reply src/style/index.css]: Hi @rithviknishad ,

I’ve updated the PR with Tailwind’s print: modifiers to handle print-specific styles instead of using traditional CSS. The changes include:

Applying padding and margin for print media.
Styling the header and footer for print.
Could you please review the changes and let me know if everything looks good or if there are any further adjustments needed?

Thanks!

File: src/style/index.css (hunk 1 above)
Comment (Zeshanxviii) [reply src/style/index.css]: Yes, you're correct. The PrintPreview component already applies padding with the p-10 class. The additional padding defined in the print styles (print:p-2) would override this padding when printing.

**Review by rithviknishad — COMMENTED** [review]
Current changes LGTM.

Did you accidentally miss out on updating Investigations Summary too, it was mentioned in the issue?

**Comment by rithviknishad** [comment]
You can see them in this route:

https://care.ohc.network/facility/0c95c7f0-e1d2-4aff-83fa-933cef60d3a8/patient/7e8fc20f-02da-47f7-8c33-f2d8c7e52dd8/consultation/edfe0bd6-0f50-4a4c-8cd0-2d4ad0e9cea1/investigation/e4759286-99a8-41a6-bc79-8628478b61d0

```
user: devdistrictadmin
pass: Coronasafe@123
```

**Comment by Zeshanxviii** [comment]
image
this is how it looks like

and printed sample
image


[InvestigationSummary.pdf](https://github.com/user-attachments/files/16826038/InvestigationSummary.pdf)

File: src/Components/Common/PageTitle.tsx [hunk 3]
```diff
@@ -70,15 +70,15 @@ export default function PageTitle({
             >
               <CareIcon
                 icon="l-angle-left"
-                className="border-box mr-1 rounded-md text-5xl hover:bg-secondary-200"
```
Comment (rithviknishad) [inline src/Components/Common/PageTitle.tsx]: Can we wrap only the necessary components under `Page`/`PageTitle` component so that this change wouldn't be necessary?

Ideally Page/PageTitle component shouldn't be present as a child of `PrintPreview` component right.

File: tailwind.config.js [hunk 4]
```diff
@@ -23,6 +23,7 @@ module.exports = {
       vs: "348px",
       ...defaultTheme.screens,
       "3xl": "1920px",
+      print: { raw: "print" },
```
Comment (rithviknishad) [inline tailwind.config.js]: what difference does this make? This is not required as of current tailwind version right, as tailwind already includes this modifier and given that it was working in existing usages so far.

File: src/Components/Common/PageTitle.tsx (hunk 3 above)
Comment (Zeshanxviii) [reply src/Components/Common/PageTitle.tsx]: Got it.

File: tailwind.config.js (hunk 4 above)
Comment (Zeshanxviii) [reply tailwind.config.js]: is it okay ? now

File: src/Components/Common/PageTitle.tsx (hunk 3 above)
Comment (Zeshanxviii) [reply src/Components/Common/PageTitle.tsx:73]: If I don't add print-hidden on line 73, the button will appear on the printed document.

image

**Comment by rithviknishad** [comment]
@Zeshanxviii could you redo for the treatment summary with the new version as it was updated in https://github.com/ohcnetwork/care_fe/pull/8295

**Review by Zeshanxviii — COMMENTED** [review]
okay

File: src/Components/Common/components/PatientDetail.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        <div className="font-medium text-secondary-800">{name}: </div>
+        {children != null ? (
+          <span className="pl-2 font-bold">{children}</span>
+        ) : (
+          <div className="h-5 w-48 animate-pulse bg-secondary-200" />
+        )}
+      </div>
+    );
+  };
+
```
Comment (rithviknishad) [inline src/Components/Common/components/PatientDetail.tsx]: 1. Is there a reason for this component to be named specific to "Patient"? It accepts any key value pair right? Consider renaming the component to something more appropriate.
2. Why is this a named export? This file only exports this one component right? Prefer default export.

File: src/Components/Facility/TreatmentSummary.tsx [hunk 6]
```diff
@@ -528,3 +540,9 @@ function InstructionsSection({ consultationData }: IInstructionsSection) {
     </>
   );
 }
+
+
+
+// Things that i added
+
+
```
Comment (rithviknishad) [inline src/Components/Facility/TreatmentSummary.tsx]:
```suggestion
```

File: src/Components/Facility/Investigations/ShowInvestigation.tsx [hunk 7]
```diff
@@ -143,25 +147,77 @@ export default function ShowInvestigation(props: any) {
...
-        isDischargedPatient={!!consultation?.discharge_date}
-        changedFields={state.changedFields}
-        handleValueChange={handleValueChange}
-        handleUpdateCancel={handleUpdateCancel}
-        handleSave={handleSubmit}
-      />
-    </div>
+          <PrintPreview title={`Investigation Report for ${patientData?.name}`}
+          >
+        <div id="section-to-print">
```
Comment (rithviknishad) [inline src/Components/Facility/Investigations/ShowInvestigation.tsx]: `PrintPreview` component already adds a relevant `id` to the printable div right? Remove the duplicate

File: src/Components/Facility/Investigations/ShowInvestigation.tsx [hunk 8]
```diff
@@ -143,25 +147,77 @@ export default function ShowInvestigation(props: any) {
...
+          
+          <div
+          className="print:w-[1100px]">
```
Comment (rithviknishad) [inline src/Components/Facility/Investigations/ShowInvestigation.tsx]: Prefer rem values over hardcoded arbitrary pixel values. This will result in different results in different browser/system zoom levels.

File: src/Components/Medicine/PrintPreview.tsx [hunk 9]
```diff
@@ -47,7 +46,7 @@ export default function PrescriptionsPrintPreview() {
       }
       disabled={!(patient && encounter && items)}
     >
-      <div className="mb-3 flex items-center justify-between p-4">
+      <div className="mb-3 flex items-center justify-between p-4 ">
```
Comment (rithviknishad) [inline src/Components/Medicine/PrintPreview.tsx]:
```suggestion
      <div className="mb-3 flex items-center justify-between p-4">
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/Components/Common/components/PatientDetail.tsx (hunk 5 above)
Comment (Zeshanxviii) [reply src/Components/Common/components/PatientDetail.tsx]: is KeyValueDetail appropriate name for this component ?

Comment (rithviknishad) [reply src/Components/Common/components/PatientDetail.tsx]: `DeferredDetail` or `AsyncDetail` would make better sense as the primary goal of this component is to show a loading state when children is null right?


Just my thoughts (feel free to do whichever you prefer):

There are many other places where "key/value" like information is represented similarly in different ways (such as a common shared loading state, rather than individual attribute level loading state and so on).

I would prefer to keep this component defined locally in both the Prescriptions print and Treatment Summary print components. This would give us better control over the design as per context. Making it generic and reusable for such a small piece is not required.

File: src/Components/Facility/Investigations/ShowInvestigation.tsx [hunk 10]
```diff
@@ -143,25 +144,96 @@ export default function ShowInvestigation(props: any) {
...
+        </PatientDetail>
+        <p className="pt-6 text-center text-xs font-medium text-secondary-700">
+          Generated on: {formatDateTime(new Date())}
+        </p>
+        <p className="pt-1 text-center text-xs font-medium text-secondary-700">
+          This is a computer generated prescription. It shall be issued to the
+          patient only after the concerned doctor has verified the content and
+          authorized the same by affixing signature.
+        </p>
+      </div>
```
Comment (rithviknishad) [inline src/Components/Facility/Investigations/ShowInvestigation.tsx]: These were not there in Investigations before right?

File: src/Components/Facility/TreatmentSummary.tsx [hunk 11]
```diff
@@ -49,50 +53,56 @@ export default function TreatmentSummary({
...
+        </PatientDetail>
+        <p className="pt-6 text-center text-xs font-medium text-secondary-700">
+          Generated on: {formatDateTime(new Date())}
+        </p>
+        <p className="pt-1 text-center text-xs font-medium text-secondary-700">
+          This is a computer generated prescription. It shall be issued to the
+          patient only after the concerned doctor has verified the content and
+          authorized the same by affixing signature.
+        </p>
+      </div>
```
Comment (rithviknishad) [inline src/Components/Facility/TreatmentSummary.tsx]: These were not there in treatment summary before right? Why was this added?

File: src/CAREUI/misc/PrintPreview.tsx [hunk 12]
```diff
@@ -11,21 +11,23 @@ type Props = {
   title: string;
 };
 
+const PRINT_BUTTON_LABEL: string = "Print";
```
Comment (rithviknishad) [inline src/CAREUI/misc/PrintPreview.tsx]: Why was this moved to a constant?

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/Components/Facility/Investigations/ShowInvestigation.tsx (hunk 10 above)
Comment (Zeshanxviii) [reply src/Components/Facility/Investigations/ShowInvestigation.tsx]: yup

File: src/Components/Facility/TreatmentSummary.tsx (hunk 11 above)
Comment (Zeshanxviii) [reply src/Components/Facility/TreatmentSummary.tsx]: my bad

**Comment by nihal467** [comment]
@Zeshanxviii is the PR ready for review

**Comment by Zeshanxviii** [comment]
> @Zeshanxviii is the PR ready for review
yes, it is ready for review

**Comment by Zeshanxviii** [comment]
> @Zeshanxviii is the PR ready for review
now , it is ready

File: src/Components/Medicine/PrintPreview.tsx (hunk 9 above)
Comment (rithviknishad) [reply src/Components/Medicine/PrintPreview.tsx]: Still unresolved

File: src/Components/Medicine/PrintPreview.tsx
Comment (rithviknishad) [inline src/Components/Medicine/PrintPreview.tsx:1]: Could you move it back to where it was? So that when squashed and merged, the author of those lines of change remains intact.

image

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/Components/Medicine/PrintPreview.tsx
Comment (Zeshanxviii) [reply src/Components/Medicine/PrintPreview.tsx:1]: okay

Comment (Zeshanxviii) [reply src/Components/Medicine/PrintPreview.tsx:1]: is it okay now

**Review by rithviknishad — APPROVED** [review]

**Comment by rithviknishad** [comment]
@Zeshanxviii lint checks are failing. could you resolve them. ensure pre-commit hook is working locally and prettier code formatter is used.

**Review by shivankacker — APPROVED** [review]

**Comment by Zeshanxviii** [comment]
> @Zeshanxviii lint checks are failing. could you resolve them. ensure pre-commit hook is working locally and prettier code formatter is used.

I think it's fixed now

**Comment by nihal467** [comment]
LGTM

File: src/Components/Facility/Investigations/ShowInvestigation.tsx [hunk 13]
```diff
@@ -143,25 +140,16 @@ export default function ShowInvestigation(props: any) {
...
-      <PageTitle
-        title={t("investigations")}
-        className="mx-3 md:mx-4"
-        crumbsReplacements={{
-          [facilityId]: { name: patientData?.facility_object?.name },
-          [patientId]: { name: patientData?.name },
-        }}
-        backUrl={`/facility/${facilityId}/patient/${patientId}/consultation/${consultationId}`}
-      />
+    <PrintPreview title={`Investigation Report for ${patientData?.name}`}>
```
Comment (khavinshankar) [inline src/Components/Facility/Investigations/ShowInvestigation.tsx]: i18n

File: src/Components/Facility/Investigations/ShowInvestigation.tsx [hunk 14]
```diff
@@ -143,25 +140,16 @@ export default function ShowInvestigation(props: any) {
...
       <InvestigationTable
-        title={`ID: ${sessionId}`}
+        title={`Investigation Report of :${patientData?.name}`}
```
Comment (khavinshankar) [inline src/Components/Facility/Investigations/ShowInvestigation.tsx]: i18n

File: src/Components/Facility/Investigations/ShowInvestigation.tsx [hunk 15]
```diff
@@ -36,8 +34,7 @@ const updateFormReducer = (state = initialState, action: any) => {
 };
 
 export default function ShowInvestigation(props: any) {
```
Comment (khavinshankar) [inline src/Components/Facility/Investigations/ShowInvestigation.tsx]: replace any with a valid type

**Review by khavinshankar — CHANGES_REQUESTED** [review]
@Zeshanxviii  LGTM, some minor changes

File: src/Components/Facility/Investigations/ShowInvestigation.tsx (hunk 15 above)
Comment (Zeshanxviii) [reply src/Components/Facility/Investigations/ShowInvestigation.tsx]: sure

Comment (Zeshanxviii) [reply src/Components/Facility/Investigations/ShowInvestigation.tsx]: is it okay , now  ?

File: src/Components/Common/PageTitle.tsx [hunk 16]
```diff
@@ -70,15 +70,15 @@ export default function PageTitle({
...
+                className="border-box mr-1 rounded-md text-5xl hover:bg-secondary-200 print:hidden"
```
Comment (rithviknishad) [inline src/Components/Common/PageTitle.tsx]: https://github.com/ohcnetwork/care_fe/pull/8441/files#r1739974547 is still not resolved.

Is this needed? We wouldn't be rendering the `PageTitle` component as child of `PrintPreview` component right? If we are, set the `print:hidden` class for the parent element instead of this reusable component.

```suggestion
                className="border-box mr-1 rounded-md text-5xl hover:bg-secondary-200"
```

File: src/Components/Facility/Investigations/ShowInvestigation.tsx [hunk 17]
```diff
@@ -141,27 +145,23 @@ export default function ShowInvestigation(props: any) {
...
-        title={t("investigations")}
-        className="mx-3 md:mx-4"
-        crumbsReplacements={{
-          [facilityId]: { name: patientData?.facility_object?.name },
-          [patientId]: { name: patientData?.name },
-        }}
-        backUrl={`/facility/${facilityId}/patient/${patientId}/consultation/${consultationId}`}
-      />
+    <PrintPreview
+      title={t("Investigation Report for {{name}}", {
```
Comment (rithviknishad) [inline src/Components/Facility/Investigations/ShowInvestigation.tsx]: Could you add this translation key to `en` locale files https://github.com/ohcnetwork/care_fe/blob/develop/src/Locale/en/Consultation.json#L25 ?

File: src/Components/Facility/Investigations/ShowInvestigation.tsx [hunk 18]
```diff
@@ -141,27 +145,23 @@ export default function ShowInvestigation(props: any) {
...
+        name: patientData?.name,
+      })}
+    >
       <InvestigationTable
-        title={`ID: ${sessionId}`}
+        title={t("Investigation Report of : {{name}}", {
+          name: patientData?.name,
+        })}
```
Comment (rithviknishad) [inline src/Components/Facility/Investigations/ShowInvestigation.tsx]: same here.

File: src/Components/Facility/Investigations/InvestigationTable.tsx [hunk 19]
```diff
@@ -173,12 +168,13 @@ export const InvestigationTable = ({
...
                       }}
+                      className="print:text-black"
                     />
                   );
                 })
               ) : (
-                <tr className="text-center text-secondary-500">
-                  No tests taken
+                <tr className="text-center text-secondary-500 print:text-black">
+                  <td className="col-span-6">No tests taken</td>
```
Comment (rithviknishad) [inline src/Components/Facility/Investigations/InvestigationTable.tsx]: Add to i18n

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/Components/Facility/Investigations/ShowInvestigation.tsx (hunk 17 above)
Comment (Zeshanxviii) [reply src/Components/Facility/Investigations/ShowInvestigation.tsx]: sure

File: src/Components/Facility/Investigations/InvestigationTable.tsx (hunk 19 above)
Comment (Zeshanxviii) [reply src/Components/Facility/Investigations/InvestigationTable.tsx]: added

File: src/Components/Common/PageTitle.tsx [hunk 20]
```diff
@@ -78,7 +78,7 @@ export default function PageTitle({
         </div>
         {componentRight}
       </div>
-      <div className={hideBack ? "my-2" : "my-2 ml-8"}>
+      <div className={hideBack ? "my-2" : "my-2 ml-8 print:hidden"}>
```
Comment (rithviknishad) [inline src/Components/Common/PageTitle.tsx]:
```suggestion
      <div className={hideBack ? "my-2" : "my-2 ml-8"}>
```

**Review by rithviknishad — APPROVED** [review]

File: src/Components/Facility/Investigations/InvestigationTable.tsx [hunk 21]
```diff
@@ -173,12 +168,13 @@ export const InvestigationTable = ({
...
                       }}
+                      className="print:text-black"
                     />
                   );
                 })
               ) : (
-                <tr className="text-center text-secondary-500">
-                  No tests taken
+                <tr className="text-center text-secondary-500 print:text-black">
+                  <td className="col-span-6">t(no_tests_taken)</td>
```
Comment (rithviknishad) [inline src/Components/Facility/Investigations/InvestigationTable.tsx]: You missed quotes and brackets here.

```suggestion
                  <td className="col-span-6">{t("no_tests_taken")}</td>
```

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]
